nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/kafka/processors/ConsumeKafka.java

```diff
@@ -513,43 +609,95 @@ public List<ConfigVerificationResult> verify(final ProcessContext context, final
+
         final KafkaConsumerService consumerService = consumerServices.poll();
         if (consumerService != null) {
             return consumerService;
         }
 
-        final int activeCount = activeConsumerCount.incrementAndGet();
-        if (activeCount > getMaxConsumerCount()) {
-            getLogger().trace("No Kafka Consumer Service available; have already reached max count of {} so will not create a new one", getMaxConsumerCount());
-            activeConsumerCount.decrementAndGet();
+        final boolean isExplicitPartitionMapping = ConsumerPartitionsUtil.isPartitionAssignmentExplicit(context.getAllProperties());
+
+        if (isExplicitPartitionMapping) {
+            getLogger().trace("No Partitioned Kafka Consumer Service available, all specified partitions are being consumed from.");
             return null;
+        } else {
+            final int activeCount = activeConsumerCount.incrementAndGet();
+            if (activeCount > getMaxConsumerCount()) {
+                getLogger().trace("No Kafka Consumer Service available; have already reached max count of {} so will not create a new one", getMaxConsumerCount());
+                activeConsumerCount.decrementAndGet();
+                return null;
+            }
+
+            getLogger().info("No Kafka Consumer Service available; creating a new one. Active count: {}", activeCount);
+            return connectionService.getConsumerService(pollingContext);
         }
+    }
 
-        getLogger().info("No Kafka Consumer Service available; creating a new one. Active count: {}", activeCount);
-        final KafkaConnectionService connectionService = context.getProperty(CONNECTION_SERVICE).asControllerService(KafkaConnectionService.class);
-        return connectionService.getConsumerService(pollingContext);
+    private int getPartitionCount(final KafkaConnectionService connectionService) {
+        Collection<String> topics = this.pollingContext.getTopics();
+
+        if (topics.isEmpty()) {
+            return -1;
+        }
+
+        int partitionsEachTopic = 0;
+        try (KafkaConsumerService kafkaConsumerService = connectionService.getConsumerService(this.pollingContext)) {
+            Map<String, List<PartitionState>> topicToPartitionStates = kafkaConsumerService.getPartitionStatesByTopic();
+            for (List<PartitionState> partitionStatesForTopic : topicToPartitionStates.values()) {
+                final int partitionsThisTopic = partitionStatesForTopic.size();
+                if (partitionsEachTopic != 0 && partitionsThisTopic != partitionsEachTopic) {
+                    throw new IllegalStateException("The specific topic names do not have the same number of partitions");
+                }
+
+                partitionsEachTopic = partitionsThisTopic;
+            }
+        } catch (IOException e) {
+            getLogger().warn("Couldn't close KafkaConsumerService after partition assignment check.", e);
```

Comment:
It is best to avoid conjunctions and avoid ending with `.` characters in messages:
```suggestion
            getLogger().warn("Failed to close KafkaConsumerService after partition assignment check", e);
```